common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -152,6 +152,10 @@ public Random current() {
         //           (note: that JDK's direct memory limit is independent of this).
         long maxDirectMemory = SystemPropertyUtil.getLong("io.netty.maxDirectMemory", -1);
 
+        if (maxDirectMemory < 0) {
+          maxDirectMemory = MAX_DIRECT_MEMORY;
+        }
+
```

Comment:
why would this solve the issue ?
the same piece of code is done in line 160

Reply:
I've removed the redundant assignment, but this needs to be here as if reflection is disabled you'll get the original error.

Reply:
I am trying to understand why do we need to move the assignment there,
is it because `PlatformDependent0.hasDirectBufferNoCleanerConstructor()` returns false ?

Because otherwise, there is no need, if `maxDirectMemory` is < 0, and we have unsafe access, then we will go to the `else` block and properly set the  `maxDirectMemory` and `USE_DIRECT_BUFFER_NO_CLEANER` flags.

Reply:
Yes, it addresses the scenario in which reflection is disabled, when either the JDK is 14 or `siren.io.netty.tryReflectionSetAccessible` is set explicitly to `false`.